RFCs/[RFC-003-RTVF]-RTA-Transaction-Validation-Flow.md

```diff
@@ -109,38 +109,45 @@ To validate RTA Transaction graftnode should perform several checks:
    
       > **TODO:** _Boris:_ I'm leaning towards trusting PoS proxy with respect to auth sample but use the PoS one-time identification keypair to enforce anonymity.
    
->**Payment Block Definition**
->
->  **Payment block** is a historical block in the blockchain, which selected by the block number as a difference between current blockchain height and constant value, which determines the delay for increasing the stability of selected auth sample (Currently we use SVP). Formally,`payment_block_number = current_block_number - SVP`. Block defined by using its block number and block hash.
+    >**Payment Block Definition**
+    >
+    >  **Payment block** is a historical block in the blockchain, which selected by the block number as a difference between current blockchain height and constant value, which determines the delay for increasing the stability of selected auth sample (Currently we use SVP). Formally,`payment_block_number = current_block_number - SVP`. Block defined by using its block number and block hash.
+
+    > **Purchase details**  is an information about purchase - may include list of pushase items, price and amount of each item, etc
+
+    > **Payment data** is a data structure containing **encrypted purchase details** and total amount  (**FullTxAmount** in atomic units)
+
+2. When PoS got data from Proxy Supernode, it prepares and sends **encrypted payment data** (to `/dapi/sale` endpoint):
+    * ~~generates the symmetric encryption key, called **PoS data encryption key**. This key will be only passed to Wallet app (only Wallet app can read this payment details)~~
+        > NOTE: for now we don't have an interface do encrypt/decrypt with symmetric keys and error handling, so it's ok to re-use `graft::crypto_tools::encryptMessage` to encrypt payment data for wallet: PoS will generate one-time keypair, encrypt and pass private key (instead of symmetric encryption key) via QR Code.
```

Comment:
yes, possible. then again, why not D.-H.?

Reply:
We can do DH but this would require some re-design as POS and Wallet can be only connected over the net using P2P messages